### ISSUE TITLE
Feature request: mention

### ISSUE THREAD

**Issue by cgestes** [memspace/zefyr#159]
Are mentions a welcome addition to Zefyr?  (to link to people or pages with @)

If that's the case do you have any idea how to proceed?

I was thinking about adding a button in the toolbar triggering an overlay using a user provided delegate to provide completion.

**Comment by cgestes** [comment]
About the underlying format, it looks like they can be implemented either using:

```@name``` or ```[@name](internal-reference)```

I feel like the second option is a bit better as it allow to store both a name and a reference (for example a user id).

**Comment by pulyaevskiy** [comment]
I agree, second option makes more sense. The name would actually be just the text itself, and the ref is a metadata associated with this text.

This is definitely something I’d like to see in Zefyr as well.

However, the way style attributes are implemented in Zefyr right now, does not allow for easy extensibility, so I’m currently looking into refactoring this part of the editor. 

That said, this discussion is still important since it helps me in understanding what kind of api interface Zefyr should provide.

In regards to mentions, I think we’ll need a notion of trigger characters, e.g. when user types @ we trigger a drop down menu with autocomplete suggestions. All of this has to be part of this new api for defining new styles and customizing behavior of the editor.

I’m still in designing phase on this, but it’s the next big thing I’ll be working on here. In fact, I’m thinking this would be the main change required for Zefyr 1.0 release.

**Comment by leedstyh** [comment]
Also, not only `@`, I need some more like `#` for topic and `$` for stock symbol. So a generic way to customize this kind of symbols (@ , #, $, and so on) is much more helpful.

**Comment by pulyaevskiy** [comment]
Exactly, that’s why I referred to these us “trigger characters” in general. New api will aim to allow registering different triggers and specify behaviors.

I know there is another request for hashtags, it will be covered similarly.

**Comment by pulyaevskiy** [comment]
There are some tricky edge cases with these though. One example is that we don’t want to trigger a mention when inside a code block, where @ is used commonly for annotations in different languages.

**Comment by YudiTan** [comment]
@cgestes might have been a while for you but is there a way to still track your progress / view your past implementation?

Any updates on this trigger character feature (is this available in v1.0 dev)? Am also trying to hack together a minimum viable implementation for this feature.
